lib/galaxy/jobs/__init__.py

```diff
@@ -1402,7 +1402,7 @@ def _collect_metrics( self, has_metrics ):
         job = has_metrics.get_job()
         per_plugin_properties = self.app.job_metrics.collect_properties( job.destination_id, self.job_id, self.working_directory )
         if per_plugin_properties:
-            log.info( "Collecting job metrics for %s" % has_metrics )
+            log.info( "Collecting job metrics for %s: %s" % ( has_metrics, getattr( has_metrics, 'id', None ) ) )
```

Comment:
I had considered removing that as well but opted for keeping it, because the has_metrics object could potentially not be a ```galaxy.model.Job``` object (a Task perhaps?). So we'd want to be able to get at least the ```type()``` of the object (which is available in the previous string representation of the Job object -- other objects could have better string representations). Same reasoning was held for the access of the 'id' attribute, as the methods have no requirement for 'id' being present in the ```has_metrics``` object.